pkg/packagekit/updates.jsx

```diff
@@ -238,68 +243,92 @@ class UpdateItem extends React.Component {
-                        </a>)
-                      )) }
-                </p>
-            );
+        var cves = null;
+        if (info.cve_urls && info.cve_urls.length) {
+            cves = insertCommas(info.cve_urls.map(url => (
+                <a href={url} rel="noopener" referrerpolicy="no-referrer" target="_blank">
+                    {url.match(/[^/=]+$/)}
+                </a>)
+            ));
         }
 
-        /* truncate long package list by default */
-        var pkgList = this.props.pkgNames.map(n => (<Tooltip tip={packageSummaries[n]}><span>{n}</span></Tooltip>));
-        var pkgs;
-        if (!this.state.expanded && pkgList.length > 15) {
-            pkgs = (
-                <div onClick={ () => this.setState({expanded: true}) }>
-                    {insertCommas(pkgList.slice(0, 15))}
-                    <a className="info-expander">{ cockpit.format(_("$0 more…"), pkgList.length - 15) }</a>
-                </div>);
+        var type;
+        if (info.severity === PK_INFO_ENUM_SECURITY) {
+            type = (
+                <span>
+                    <span className="pficon pficon-security">&nbsp;</span>
+                    { (info.cve_urls && info.cve_urls.length > 0) ? info.cve_urls.length : "" }
+                </span>);
+        } else if (info.severity >= PK_INFO_ENUM_NORMAL) {
+            type = (
+                <span>
+                    <span className="fa fa-bug">&nbsp;</span>
+                    { bugs ? info.bug_urls.length : "" }
+                </span>);
         } else {
-            pkgs = insertCommas(pkgList);
+            type = (
+                <span>
+                    <span className="pficon pficon-enhancement">&nbsp;</span>
+                    { bugs ? info.bug_urls.length : "" }
+                </span>);
         }
 
-        /* truncate long description by default */
-        var descLines = (info.description || "").trim().split("\n");
-        var desc;
-        if (!this.state.expanded && descLines.length > 7) {
-            desc = (
-                <div onClick={ () => this.setState({expanded: true}) }>
-                    {descLines.slice(0, 6).join("\n") + "\n"}
-                    <a>{_("More information…")}</a>
-                </div>);
-        } else {
-            desc = info.description;
+        var pkgList = this.props.pkgNames.map(n => (<Tooltip tip={packageSummaries[n]}><span>{n}</span></Tooltip>));
+        var pkgs = insertCommas(pkgList);
+        var pkgsTruncated = pkgs;
+        if (pkgList.length > 4)
+            pkgsTruncated = insertCommas(pkgList.slice(0, 4).concat("…"));
+
+        var descriptionFirstLine = (info.description || "").trim();
+        if (descriptionFirstLine.indexOf("\n") >= 0)
+            descriptionFirstLine = descriptionFirstLine.slice(0, descriptionFirstLine.indexOf("\n"));
+
+        var details = null;
+        if (this.state.expanded) {
+            details = (
+                <tr className="listing-ct-panel">
```

Comment:
Why is this not `<ListingRow>` from `lib/cockpit-components-listing.jsx`?

Reply:
Mostly because that currently insists on putting tabs into the panel, and we don't have/want to use any. That can of course be made more flexible, but I didn't want to block on this for now.